app/models/user.rb

```diff
@@ -0,0 +1,25 @@
+class User < ApplicationRecord
+  validates :name, presence: true
+  validates :photo, presence: true
+  validates :post_counter, presence: false
+
+  has_many :comments
+  has_many :likes
+  has_many :posts
+  # retrieve all the comments that the user has liked
+  has_many :liked_comments, through: :likes, source: :comment
+  # retrieve all the posts that the user has liked
+  has_many :liked_posts, through: :posts, source: :post
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_post_counter
+    update(post_counter: posts.count)
+  end
```

Comment:
- [ ] According to the project requirements, this method to update the posts counter for a user should be in the `Post` model instead of here. Please rectify this 👍 

**HINT:** In the `Post` model, you can take advantage of the relationship between `author` and `posts` like this 👇
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```

Reply:
Thanks alot @codepantha